test/e2e/tests/e2e_test.go

```diff
@@ -155,3 +161,68 @@ func fetchBlockChain(t *testing.T) []*types.Block {
 
 	return blocks
 }
+
+// fetchABCIRequests go through the logs of a specific node and collect all ABCI requests (each slice represents requests from beggining until the first crash,
+// and then between two crashes) for a specific node.
+func fetchABCIRequests(t *testing.T, nodeName string) ([][]*abci.Request, error) {
+	testnet := loadTestnet(t)
+	logs, err := fetchNodeLogs(testnet, nodeName)
+	if err != nil {
+		return nil, err
+	}
+	reqs := make([][]*abci.Request, 0)
+	i := -1
+	// Parse output line by line.
+	lines := strings.Split(string(logs), "\n")
+	for _, line := range lines {
+		if strings.Contains(line, "Application started") {
+			i += 1
+			reqs = append(reqs, make([]*abci.Request, 0))
+			continue
+		}
+		r, err := app.GetABCIRequestFromString(line)
+		require.NoError(t, err)
+		// Ship the lines that does not contain abci request.
+		if r == nil {
+			continue
+		}
+		reqs[i] = append(reqs[i], r)
```

Comment:
```suggestion
		reqs[len(reqs)-1] = append(reqs[len(reqs)-1], r)
```
This allows you to get rid of `i` and is more readable IMO

Reply:
True! Thanks!